packages/web3/src/v3/WalletLink.ts

```diff
@@ -25,31 +25,27 @@ export class WalletLink {
         }
     }
 
-    private async assertNotAlreadyLinked(rootKey: ethers.Signer, wallet: ethers.Signer | Address) {
-        const rootKeyAddress = await rootKey.getAddress()
+    public async isLinkedAlready(walletAddress: string): Promise<boolean> {
+        const rootKeyAddress = await this.walletLinkShim.read.getRootKeyForWallet(walletAddress)
+
+        return rootKeyAddress !== INVALID_ADDRESS
+    }
+
+    private async assertNotAlreadyLinked(wallet: ethers.Signer | Address) {
         const walletAddress = typeof wallet === 'string' ? wallet : await wallet.getAddress()
-        const isLinkedAlready = await this.walletLinkShim.read.checkIfLinked(
-            rootKeyAddress,
-            walletAddress,
-        )
 
-        if (isLinkedAlready) {
+        if (await this.isLinkedAlready(walletAddress)) {
             throw new WalletAlreadyLinkedError()
         }
 
-        return { rootKeyAddress, walletAddress }
+        return { walletAddress }
```

Comment:
kinda weird that these return the wallet address
And I think we can drop the "already" from these function names